adapters/httpapi_curl.c

```diff
@@ -401,6 +401,31 @@ HTTPAPI_RESULT HTTPAPI_ExecuteRequest(HTTP_HANDLE handle, HTTPAPI_REQUEST_TYPE r
 
                     break;
 
+                case HTTPAPI_REQUEST_HEAD:
+                    if (curl_easy_setopt(httpHandleData->curl, CURLOPT_HTTPGET, 1L) != CURLE_OK)
+                    {
+                        result = HTTPAPI_SET_OPTION_FAILED;
+                        LogError("(result = %s)", ENUM_TO_STRING(HTTPAPI_RESULT, result));
+                    }
+                    else
```

Comment:
Instead of the 
else 
{
    if (...)
[...]

Could you please change this to be more of a block of

else if (doFoo() != OK) 
{ 
  ...
}
else if (doBar()) !+ OK) 
{ 
 
}

I know this isn't current pattern in this file but its what rest of SDK is moving to since its way less indentation and easier to follow code flow.